aten/src/ATen/ExpandUtils.h

```diff
@@ -45,33 +46,81 @@ inline void check_defined(std::initializer_list<std::reference_wrapper<const Ten
   }
 }
 
-inline std::tuple<Tensor> expand_inplace(const Tensor &tensor, const Tensor &to_expand) {
+inline c10::MaybeOwned<Tensor> expand_inplace_v2(const Tensor& tensor, const Tensor& to_expand) {
   if (tensor.sizes().equals(to_expand.sizes())) {
-    return std::make_tuple(to_expand);
+    return c10::MaybeOwned<Tensor>::borrowed(to_expand);
   }
+  return c10::MaybeOwned<Tensor>::owned(to_expand.expand(tensor.sizes(), /*implicit=*/true)); // see [expand implicit]
+}
+
+// expand_inplace_v2 may borrow from to_expand; prevent calling with an
+// obvious temporary.
+inline c10::MaybeOwned<Tensor> expand_inplace_v2(const Tensor& tensor, Tensor&& to_expand) = delete;
 
-  return std::make_tuple(to_expand.expand(tensor.sizes(), /*implicit=*/true)); // see [expand implicit]
+
+C10_DEPRECATED_MESSAGE("expand_inplace is deprecated. Use expand_inplace_v2 instead.")
+inline std::tuple<Tensor> expand_inplace(const Tensor &tensor, const Tensor &to_expand) {
+  return std::make_tuple(*expand_inplace_v2(tensor, to_expand));
 }
```

Comment:
IMO let's just YOLO remove it

Reply:
OK, but then we need feature:bc-breaking, right?